odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/handlers/SaleOrderLineHandler.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.handlers;
+
+import static java.util.Arrays.asList;
+
+import com.ozonehis.eip.odoo.openmrs.Constants;
+import com.ozonehis.eip.odoo.openmrs.client.OdooClient;
+import com.ozonehis.eip.odoo.openmrs.client.OdooUtils;
+import com.ozonehis.eip.odoo.openmrs.mapper.odoo.SaleOrderLineMapper;
+import com.ozonehis.eip.odoo.openmrs.model.Product;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrder;
+import com.ozonehis.eip.odoo.openmrs.model.SaleOrderLine;
+import com.ozonehis.eip.odoo.openmrs.model.Uom;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.ProducerTemplate;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Resource;
+import org.hl7.fhir.r4.model.ServiceRequest;
+import org.openmrs.eip.EIPException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Setter
+@Component
+public class SaleOrderLineHandler {
+
+    @Autowired
+    private OdooClient odooClient;
+
+    @Autowired
+    private ProductHandler productHandler;
+
+    @Autowired
+    private UomHandler uomHandler;
+
+    @Autowired
+    private SaleOrderLineMapper<Resource> saleOrderLineMapper;
+
+    public SaleOrderLine buildSaleOrderLineIfProductExists(Resource resource, SaleOrder saleOrder) {
+        Product product = productHandler.getProduct(resource);
+        log.debug("SaleOrderLineHandler: Fetched Product {}", product);
+        if (product == null) { // TODO: Check should we allow product not found
+            log.debug("SaleOrderLineHandler: No product found");
```

Comment:
Shouldn't you throw an exception in this case?

Reply:
Leaving it as it is.